### PR TITLE
Authentikos: integration tests

### LINKED ISSUE #2139: Authentikos: unit and integration test coverage
Add unit and integration test coverage for [Authentikos](https://github.com/istio/test-infra/tree/master/authentikos) so that downstream users can consume it with confidence it will not break.

### PR DESCRIPTION
**Pull request by clarketm**
fix: https://github.com/istio/test-infra/issues/2139
for: #2511 

0b00f956fa92ff8494a1f39dc4bc5386db2a62af: Adds integration test(s)
3560ad1c2491f5c7b9998f29d15bd787eeab3b57: Adds jobs for executing both _unit_ and _integration_ tests.

### CODE REVIEW

File: authentikos/test/integ-simple.sh
Comment (howardjohn) [inline authentikos/test/integ-simple.sh]: If these fail I suspect there will be no useful error message?

File: prow/config/jobs/test-infra.yaml
Comment (howardjohn) [inline prow/config/jobs/test-infra.yaml:21]: I think this is covered by `test` job already

File: prow/config/jobs/test-infra.yaml
Comment (howardjohn) [inline prow/config/jobs/test-infra.yaml:30]: nit: the regex is fairly complicated, can you comment what we are matching and why?

**Review by howardjohn — APPROVED** [review]
/hold

File: prow/config/jobs/test-infra.yaml
Comment (clarketm) [reply prow/config/jobs/test-infra.yaml:21]: Because **Authentikos** is a submodule: 
#### go test ./... from the repository root will no longer test everything in the repository

https://github.com/golang/go/wiki/Modules#should-i-have-multiple-modules-in-a-single-repository

**e.g.**
```console
$ cd ~/SourceCode/istio_test-infra
$ go test ./... | grep authentikos; echo $?
1
```

File: authentikos/test/integ-simple.sh
Comment (clarketm) [reply authentikos/test/integ-simple.sh]: That is correct. Let me see what I can change (e.g. `set -x`, `echo "..."`) to make failures more obvious.

Comment (clarketm) [reply authentikos/test/integ-simple.sh]: I added more verbosity in the tests using `set -x` and echoing test name.

**Review by howardjohn — APPROVED** [review]

**Comment by clarketm** [comment]
ping. Can I get approval from codeowner of test-infra jobs?

**Review by cjwagner — APPROVED** [review]

**Pull request merged**
